tests/integration/conftest.py

```diff
@@ -64,4 +83,60 @@ def time_series(request, cognite_client):
     create_subscription_in_cdf(timeseries_set, sub_name, cognite_client)
     yield timeseries_set
     remove_time_series_data(timeseries_set, cognite_client)
-    cognite_client.time_series.subscriptions.delete(sub_name)
\ No newline at end of file
+    cognite_client.time_series.subscriptions.delete(sub_name)
+
+@pytest.fixture(scope="session")
+def test_space(test_config, cognite_client: CogniteClient):
+    space_id = test_config["data_modeling"][0]["space"]
+    space = cognite_client.data_modeling.spaces.retrieve(space_id)
+    if space is None:
+        new_space = SpaceApply(space_id, name="Integration Test Space", description="The space used for integration tests.")
+        space = cognite_client.data_modeling.spaces.apply(new_space)
+    else: # Ensure there aren't existing data models
+        data_model_list = cognite_client.data_modeling.data_models.list(space=space_id)
+        assert len(data_model_list) == 0, "Space should not have existing data models, please remove models/views/containers before testing"
+    yield space
+    cognite_client.data_modeling.spaces.delete(spaces=[space_id])
+    
+
+@pytest.fixture(scope="session")
+def test_dml():
```

Comment:
My reasoning for splitting out the fixtures in this way was because one test is for creation and the other is for updates, so I didn't want to apply the data model in the creation test as a fixture but in the actual test.  However, we could just focus on instance creation to make the fixtures simpler and easier to follow.  Is there anything written to Fabric before the instances are applied?

Reply:
I see. In that case it does make sense to separate it out if the tests for update use a different set of data.  I don't think anything needs to be written to Fabric upfront.

Reply:
Clarified offline that data isn't populated in Fabric unless there are instances.  Will simplify fixtures.